de.dashup.application/src/main/java/de/dashup/application/controllers/WorkbenchController.java

```diff
@@ -0,0 +1,278 @@
+                                                @RequestParam(value = "shortDescription") String shortDescription,
+                                                @RequestParam(value = "description") String description,
+                                                @RequestParam(value = "category") String category,
+                                                @RequestParam(value = "tags") String tagsJSON) throws SQLException, UnsupportedEncodingException {
+        User user = LocalStorage.getInstance().getUser(request, token);
+        if (user != null) {
+            Widget widget = new Widget();
+            updateInformation(widget, publishedId, name, shortDescription, description, category, tagsJSON);
+            return "redirect:/workbench/published/" + publishedId + "#changedInformation";
+        }
+        return "redirect:/login";
+    }
+
+    private void updateInformation(Widget widget, int publishedId, String name, String shortDescription, String description, String category, String tagsJSON) throws UnsupportedEncodingException, SQLException {
+        widget.setId(publishedId);
+        widget.setName(name);
+        widget.setShortDescription(shortDescription);
+        widget.setDescription(description);
+        widget.setCategory(category);
+
+        JSONArray json = new JSONArray(URLDecoder.decode(tagsJSON, StandardCharsets.UTF_8.name()));
+        for (int i = 0; i < json.length(); i++) {
+            widget.getTags().add(new Tag(json.getJSONObject(i).getInt("id"), json.getJSONObject(i).getString("name")));
+        }
+
+        DashupService.getInstance().updateWidgetInformation(widget);
+    }
+
+    @RequestMapping(value = "/draft/{draftId}/changeCode", method = RequestMethod.POST)
+    public String handleChangeDraftCode(@CookieValue(name = "token", required = false) String token,
+                                        HttpServletRequest request,
+                                        @PathVariable(value = "draftId") int draftId,
+                                        @RequestParam(value = "code_small", required = false) String codeSmall,
+                                        @RequestParam(value = "code_medium", required = false) String codeMedium,
+                                        @RequestParam(value = "code_large", required = false) String codeLarge) throws SQLException, UnsupportedEncodingException {
+        User user = LocalStorage.getInstance().getUser(request, token);
+        if (user != null) {
+            Draft draft = new Draft();
+            updateCode(draftId, codeSmall, codeMedium, codeLarge, draft);
+            return "redirect:/workbench/draft/" + draftId + "#changedCode";
+        }
+        return "redirect:/login";
+    }
+
+    @RequestMapping(value = "/published/{publishedId}/changeCode", method = RequestMethod.POST)
+    public String handleChangeWidgetCode(@CookieValue(name = "token", required = false) String token,
```

Comment:
You know what to do. Goes for the rest of the controller handlers as well. I will not comment this issue further.